core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -2290,10 +2290,9 @@ spliceExp = spliceWith (\l1 l2 -> max (l1 * 2) (l1 + l2))
 {-# INLINE breakOn #-}
 breakOn :: MonadIO m
     => Word8 -> MutArray Word8 -> m (MutArray Word8, Maybe (MutArray Word8))
-breakOn sep arr@MutArray{..} = asPtrUnsafe arr $ \p -> liftIO $ do
-    -- XXX Instead of using asPtrUnsafe (pinning memory) we can pass unlifted
-    -- Addr# to memchr and it should be safe (from ghc 8.4).
```

Comment:
Aha. How?

Reply:
Read: https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/ffi.html?highlight=ffi#guaranteed-call-safety

GHC, since version 8.4, guarantees that garbage collection will never occur during an unsafe call, even in the bytecode interpreter, and further guarantees that unsafe calls will be performed in the calling thread. Making it safe to pass heap-allocated objects to unsafe functions.